java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -155,7 +155,21 @@ private static boolean isMethodInvocation(@Nullable Tree tree, LambdaExpressionT
         }
         arguments = ((NewClassTree) tree).arguments();
       } else {
-        arguments = ((MethodInvocationTree) tree).arguments();
+        MethodInvocationTree mit = (MethodInvocationTree) tree;
+        if(mit.methodSelect().is(Tree.Kind.MEMBER_SELECT)) {
+          MemberSelectExpressionTree mse = (MemberSelectExpressionTree) mit.methodSelect();
+          while (mse != null) {
+            if(mse.expression().is(Tree.Kind.METHOD_INVOCATION)) {
+              return false;
```

Comment:
While I understand what are we doing here, it really took me a while to get it and I find the name of the method confusing. The while loop could be perhaps extracted to another aptly named method to make it bit more clear.